From 1efee15daaaf7c7c3ea7f9fd6d5c8ec8605a09c7 Mon Sep 17 00:00:00 2001
From: Ariadne Conill <ariadne@dereferenced.org>
Date: Thu, 3 Sep 2020 17:50:26 -0600
Subject: [PATCH] compiler.h: ensure IOPortBase is declared extern on mips

This is needed to fix compilation with -fcommon on GCC, which
is the default in GCC 10.

Signed-off-by: Ariadne Conill <ariadne@dereferenced.org>
---
 hw/xfree86/common/compiler.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/hw/xfree86/common/compiler.h b/hw/xfree86/common/compiler.h
index 2b2008b3f..c7d617eb1 100644
--- a/hw/xfree86/common/compiler.h
+++ b/hw/xfree86/common/compiler.h
@@ -525,7 +525,7 @@ xf86WriteMmio32Le(__volatile__ void *base, const unsigned long offset,
 #define PORT_SIZE short
 #endif
 
-_X_EXPORT unsigned int IOPortBase;      /* Memory mapped I/O port area */
+extern _X_EXPORT unsigned int IOPortBase;      /* Memory mapped I/O port area */
 
 static __inline__ void
 outb(unsigned PORT_SIZE port, unsigned char val)
-- 
2.28.0

